test/unit/test_runner_config.py

```diff
@@ -1,17 +1,11 @@
 # -*- coding: utf-8 -*-
 
+from functools import partial
+from io import StringIO
 import os
 import re
-import shlex
-
-try:
-    from StringIO import StringIO
-except ImportError:
-    from io import StringIO
```

Comment:
A lot of the import changes here are non-functional. Did this change to StringIO affect functionality?

Reply:
I was curious about this also, but it does pass all tests so...?

Reply:
`io.StringIO` seems to behave a bit more consistently, and checks the type and raises an error if bytes are passed in, even under Python 2.

Reply:
and if we are to deal with problems like this, I feel very strongly that we need to be more conscious of the string type being used in these tests.